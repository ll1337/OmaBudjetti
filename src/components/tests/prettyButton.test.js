import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react-native';
import PrettyButton from '../prettyButton';

describe('prettyButton', () => {

    test('renders correctly without parameters', () => {
        const tree = renderer.create(<PrettyButton />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('renders correctly with a title', () => {
        const tree = renderer.create(<PrettyButton title='Title' />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('renders correctly with a title and icons', () => {
        const tree = renderer.create(<PrettyButton
            title='Title'
            iconLeft="pulse"
            iconRight="play"
        />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('clicking the PrettyButton should call the function parameter', () => {

        // Setup
        const testFunction = jest.fn();
        render(<PrettyButton onPress={testFunction} title='Title' />);

        // Simulate user action
        fireEvent.press(screen.getByText('Title'));

        // Expect
        expect(testFunction.mock.calls).toHaveLength(1);
    });
});
