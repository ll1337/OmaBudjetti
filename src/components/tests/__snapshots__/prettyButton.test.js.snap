// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prettyButton renders correctly with a title 1`] = `
<View
  style={
    {
      "borderRadius": 60,
      "elevation": 4,
      "marginLeft": 30,
      "marginRight": 30,
      "overflow": "hidden",
      "shadowColor": "rgba(0, 0, 0, 1)",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 4,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      {
        "colors": [
          3474437549,
          3645748977,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      {
        "borderRadius": 60,
        "bottom": -2,
        "left": -2,
        "position": "absolute",
        "right": -2,
        "top": -2,
      }
    }
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F8F9F9",
        "borderRadius": 60,
        "borderWidth": 0,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 2,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 2,
        "opacity": 1,
        "overFlow": "hidden",
      }
    }
  >
    <View
      color="#000000"
      style={
        [
          {
            "backgroundColor": "rgba(255,255,255,0)",
            "borderRadius": 25,
            "height": 50,
            "width": 50,
          },
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "color": "#000000",
          },
        ]
      }
    />
    <Text
      style={
        {
          "fontFamily": "Roboto",
          "fontSize": 22,
          "textAlign": "center",
        }
      }
    >
      Title
    </Text>
    <View
      color="#000000"
      style={
        [
          {
            "backgroundColor": "rgba(255,255,255,0)",
            "borderRadius": 25,
            "height": 50,
            "width": 50,
          },
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "color": "#000000",
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`prettyButton renders correctly with a title and icons 1`] = `
<View
  style={
    {
      "borderRadius": 60,
      "elevation": 4,
      "marginLeft": 30,
      "marginRight": 30,
      "overflow": "hidden",
      "shadowColor": "rgba(0, 0, 0, 1)",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 4,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      {
        "colors": [
          3474437549,
          3645748977,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      {
        "borderRadius": 60,
        "bottom": -2,
        "left": -2,
        "position": "absolute",
        "right": -2,
        "top": -2,
      }
    }
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F8F9F9",
        "borderRadius": 60,
        "borderWidth": 0,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 2,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 2,
        "opacity": 1,
        "overFlow": "hidden",
      }
    }
  >
    <View
      color="#000000"
      style={
        [
          {
            "backgroundColor": "rgba(255,255,255,0)",
            "borderRadius": 25,
            "height": 50,
            "width": 50,
          },
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "color": "#000000",
          },
        ]
      }
    >
      <Text />
    </View>
    <Text
      style={
        {
          "fontFamily": "Roboto",
          "fontSize": 22,
          "textAlign": "center",
        }
      }
    >
      Title
    </Text>
    <View
      color="#000000"
      style={
        [
          {
            "backgroundColor": "rgba(255,255,255,0)",
            "borderRadius": 25,
            "height": 50,
            "width": 50,
          },
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "color": "#000000",
          },
        ]
      }
    >
      <Text />
    </View>
  </View>
</View>
`;

exports[`prettyButton renders correctly without parameters 1`] = `
<View
  style={
    {
      "borderRadius": 60,
      "elevation": 4,
      "marginLeft": 30,
      "marginRight": 30,
      "overflow": "hidden",
      "shadowColor": "rgba(0, 0, 0, 1)",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 4,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      {
        "colors": [
          3474437549,
          3645748977,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      {
        "borderRadius": 60,
        "bottom": -2,
        "left": -2,
        "position": "absolute",
        "right": -2,
        "top": -2,
      }
    }
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F8F9F9",
        "borderRadius": 60,
        "borderWidth": 0,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 2,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 2,
        "opacity": 1,
        "overFlow": "hidden",
      }
    }
  >
    <View
      color="#000000"
      style={
        [
          {
            "backgroundColor": "rgba(255,255,255,0)",
            "borderRadius": 25,
            "height": 50,
            "width": 50,
          },
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "color": "#000000",
          },
        ]
      }
    />
    <Text
      style={
        {
          "fontFamily": "Roboto",
          "fontSize": 22,
          "textAlign": "center",
        }
      }
    />
    <View
      color="#000000"
      style={
        [
          {
            "backgroundColor": "rgba(255,255,255,0)",
            "borderRadius": 25,
            "height": 50,
            "width": 50,
          },
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "color": "#000000",
          },
        ]
      }
    />
  </View>
</View>
`;
